extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.events.js

```diff
@@ -0,0 +1,117 @@
+define('wikia.articleVideo.featuredVideo.events', function () {
+	var state = getNewState(),
+		wasAlreadyUnmuted = false,
+		depth = 0,
+		wasStartEventFired = false,
+		prefixes = {
+			ad: 'ad',
+			video: 'video'
+		},
+		playerInstance;
+
+	function getDefaultState() {
+		return {
+			wasFirstQuartileTriggered: false,
+			wasMidPointTriggered: false,
+			wasThirdQuartileTriggered: false,
+			progress: {
+				durationWatched: 0,
+				percentWatched: 0
+			}
+		}
+	}
+
+	function getNewState() {
+		return {
+			ad: getDefaultState(),
+			video: getDefaultState()
+		}
+	}
+
+	function handleTime(prefix, data) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)